pkgs/development/python-modules/backports-zoneinfo/default.nix

```diff
@@ -16,19 +18,31 @@ buildPythonPackage rec {
     sha256 = "sha256-00xdDOVdDanfsjQTd3yjMN2RFGel4cWRrAA3CvSnl24=";
   };
 
-  propagatedBuildInputs = lib.optionals (pythonOlder "3.7") [
+  propagatedBuildInputs = [
+    tzdata
+  ] ++ lib.optionals (pythonOlder "3.7") [
     importlib-resources
   ];
 
   pythonImportsCheck = [ "backports.zoneinfo" ];
 
+  preCheck = ''
+    export PYTHONTZPATH=${pkgs.tzdata}/share/zoneinfo
```

Comment:
Wasn't there some change in python 3.10 that is not compatible with this? Can't remember the details right now.

Reply:
tzdata is hardcoded in 3.10. Before 3.10 Python is not aware of `PYTHONTZPATH`. `backports.zoneinfo` probably needs to be patched to hardcode the path to `tzdata`.

Reply:
Does this look better? https://github.com/NixOS/nixpkgs/pull/125440/commits/ec68123b8419a2fb9dab9c77106fb1e42b5f4663

Reply:
Moved into a separate pr against master, because I might need this earlier than expected.

https://github.com/NixOS/nixpkgs/pull/125599